src/compiler/checker.ts

```diff
@@ -18420,6 +18444,46 @@ namespace ts {
             return checkPropertyAccessExpressionOrQualifiedName(node, node.left, node.right);
         }
 
+        function getPropertyByPrivateName(apparentType: Type, leftType: Type, right: PrivateName): Symbol | undefined {
+            let classWithShadowedPrivateName;
+            let klass = getContainingClass(right);
+            while (klass) {
+                const symbolTableKey = getPropertyNameForPrivateNameDescription(klass.symbol, right.escapedText);
+                if (symbolTableKey) {
+                    const prop = getPropertyOfType(apparentType, symbolTableKey);
+                    if (prop) {
+                        if (classWithShadowedPrivateName) {
+                            error(

```

Comment:
We don't generally report errors in `get`-named functions without a means to opt-out (for fast-path checks, API calls, tight loops, etc.) as we don't want to add a lot of extraneous diagnostics we have to deduplicate later. For example, `getIteratedType` takes an `errorNode` that may be `undefined` if we don't need to report a diagnostic. If a case need to report an error, it passes the error node, otherwise it passes `undefined`. This would allow us to add this function to the public `TypeChecker` API to allow API consumers to attempt lookups without incurring additional memory overhead.